### PR TITLE
interop with js-ipfs-mdns

### PR DESCRIPTION
**Pull request by richardschneider**
Hi there, great project. 

I'm trying to get the service address(es) for [js-ipfs-mdns](https://github.com/libp2p/js-libp2p-mdns) and have run into the following issues. Please review my PR and make any **suggestions** you want.

#### It's service name does not end with `.local.`

I consider this a bug of  `js-libp2p-mdns`.  ~~I've made a work-around to your code to bypass appending '.local.' to a service tag when it ends with '.'~~

####   It sometimes exists only on the local host's loopback

Remove the code that ignores the loopback check

#### It does not set the the AA bit in the response

According to [RFC6762](https://tools.ietf.org/html/rfc6762#section-18.4)

> the Authoritative Answer  bit ... MUST be  ignored on reception.

### CODE REVIEW

File: src/Tmds.MDns/NetworkInterfaceHandler.cs [hunk 1]
```diff
@@ -239,8 +239,8 @@ private void OnReceive(IAsyncResult ar)
                                 }
                             }
                         }
-                    }
-                    if (header.IsResponse && header.IsNoError && header.IsAuthorativeAnswer)
+                    }
+                    if (header.IsResponse && header.IsNoError)
```
Comment (tmds) [inline src/Tmds.MDns/NetworkInterfaceHandler.cs]: :+1:

File: src/Tmds.MDns/ServiceBrowser.cs [hunk 2]
```diff
@@ -263,7 +259,18 @@ private void CheckNetworkInterfaceStatuses(Dictionary<int, NetworkInterfaceHandl
...
+                        // is appended.  However, some services do not follow this convention.  So
+                        // if the service type ends with ".", it is considered a FQN.
+                        var names = _serviceTypes.Select(st =>
+                        {
+                            var fqn = st.ToLowerInvariant();
+                            if (!fqn.EndsWith("."))
+                                fqn = fqn + ".local.";
+                            return new Name(fqn);
+                        });
+                        interfaceHandler.StartBrowse(names);
```
Comment (tmds) [inline src/Tmds.MDns/ServiceBrowser.cs]: We shouldn't workaround js-libp2p-mdns behavior. It should be fixed in that project. Please remove this.

**Comment by tmds** [comment]
hi @richardschneider thank you for suggesting these improvements. I've added some comments. PTAL.

File: src/Tmds.MDns/ServiceBrowser.cs (hunk 2 above)
Comment (richardschneider) [reply src/Tmds.MDns/ServiceBrowser.cs]: Would prefer to keep it...but will remove it.

**Comment by richardschneider** [comment]
@tmds Anything else?

**Comment by tmds** [comment]
I think you added more stuff than you intended: https://github.com/tmds/Tmds.MDns/pull/18/commits/82a59b7758d215d857f24d8b9a7b6c3926dbeba8

**Comment by richardschneider** [comment]
oops...its late (Friday 2300 in NZ)...sorry...will fix

**Comment by richardschneider** [comment]
@tmds should be ready to go!

**Comment by richardschneider** [comment]
@tmds Any comments?

**Comment by tmds** [comment]
LGTM! Thank you @richardschneider !

**Comment by tmds** [comment]
Version 0.7.0 is on nuget.org that includes these changes.

**Comment by richardschneider** [comment]
And thank you for the library.

**Pull request merged**